doc/_extensions/options_from_kconfig.py

```diff
@@ -40,20 +41,102 @@ def _monkey_patch_kconfiglib(kconfiglib):
+            return build_path
+
+        rel_part = parts[-1].lstrip(os.sep)
+
+        # Remove common build subdirectories
+        for prefix in ['nrf/src/', 'nrf/']:
+            if rel_part.startswith(prefix):
+                rel_part = rel_part[len(prefix):]
+                break
+
+        return os.path.join(self.config.options_from_kconfig_base_dir, rel_part)
+
+    def _get_source_directory(self):
+        """
+        Get the directory of the current RST file.
+
+        Returns:
+            Absolute path to the directory containing the RST file
+        """
+        source = self.state_machine.input_lines.source(
+            self.lineno - self.state_machine.input_offset - 1)
+        return os.path.dirname(os.path.abspath(source))
+
+    def _strip_quotes(self, text):
+        """
+        Remove surrounding quotes from text if present.
+
+        Args:
+            text: String that may have surrounding quotes
+
+        Returns:
+            String with quotes removed if they were present
+        """
+        if text and len(text) >= 2:
+            if (text.startswith('"') and text.endswith('"')) or \
+               (text.startswith("'") and text.endswith("'")):
+                return text[1:-1]
+        return text
 
     def run(self):
+        source_dir = self._get_source_directory()
+
         if len(self.arguments) > 0:
-            _, path = self.env.relfn2path(self.arguments[0])
+            # Case 1: Explicit path argument provided
+            arg_path = self.arguments[0]
+
+            if os.path.isabs(arg_path):
+                path = arg_path
+            else:
+                # Calculate relative path from sphinx source directory
+                rel_source_path = os.path.relpath(source_dir, self.env.srcdir)
+                # Combine with base directory and argument path
+                path = os.path.normpath(os.path.join(
+                    self.config.options_from_kconfig_base_dir,
+                    rel_source_path,
+                    arg_path
+                ))
+
+                # Fallback: try other resolution methods if file doesn't exist
+                if not os.path.exists(path):
+                    # Convert build path to source path and try direct resolution
+                    converted_source_dir = self._convert_build_to_source_path(source_dir)
+                    fallback_path = os.path.normpath(os.path.join(converted_source_dir, arg_path))
+                    if os.path.exists(fallback_path):
+                        path = fallback_path
+                    else:
+                        # Additional fallback: try using the old relfn2path method
+                        try:
+                            _, old_path = self.env.relfn2path(arg_path)
+                            if os.path.exists(old_path):
+                                path = old_path
+                        except Exception:
+                            pass  # Keep the original path for error reporting
```

Comment:
can you explain when is this fallback needed?

Reply:
This is to solve cases where the base path would conflict with the relative path provided in the tag, such as in this case:
```
Kconfig file not found: /home/runner/work/sdk-nrf/sdk-nrf/ncs/nrf/doc/_build/nrf/src/applications/nrf5340_audio/src/modules/Kconfig
```
So the aim is to look for the file in the source, not in the build.